framework/src/JovoDevice.ts

```diff
@@ -0,0 +1,47 @@
+import { JovoRequest } from './JovoRequest';
+import { JovoResponse } from '@jovotech/output';
+
+import { Jovo } from './Jovo';
+
+export type Capability = 'screen' | 'audio' | 'long-form-audio' | string;
+
+export type JovoDeviceConstructor<
+  REQUEST extends JovoRequest,
+  RESPONSE extends JovoResponse,
+  JOVO extends Jovo<REQUEST, RESPONSE>,
+  CAPABILITY extends Capability = Capability,
+> = new (jovo: JOVO) => JovoDevice<REQUEST, RESPONSE, JOVO, CAPABILITY>;
+
+export abstract class JovoDevice<
+  REQUEST extends JovoRequest,
+  RESPONSE extends JovoResponse,
+  JOVO extends Jovo<REQUEST, RESPONSE>,
+  CAPABILITY extends Capability = Capability,
+> {
+  capabilities: CAPABILITY[] = [];
+
+  public constructor(readonly jovo: JOVO) {
+    this.setCapabilitiesFromRequest();
+  }
+
+  protected abstract setCapabilitiesFromRequest(): void;
+
+  // TODO: AND or OR ? currently OR
```

Comment:
I'm not sure what you mean here, but if you mean `AND` in the sense of the Intersection Type I think this wouldn't really work here.

Reply:
I also think that the feature can be omitted because developers can simply use `supports` multiple times. That way developers have much more control as well. In complex situations, helper functions could be implemented.